### PR TITLE
 Plant test for this release to continue UAT/testnet chain 

### PR DESCRIPTION
**Pull request by lucafavatella**
Delivers:
* https://www.pivotaltracker.com/story/show/156846658
* https://www.pivotaltracker.com/story/show/156724656

Depends on PR #1014 (included in this PR).

TODO:
* [x] Re-enable spend tx smoke test
* [x] Factor common code of the two spend tx smoke tests
* [ ] Attempt to reduce instances of protocol versions in repo
  * I will address this in separate PR @eproxus 
* [x] Rebase on master

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -155,16 +155,20 @@ system-test:
 	@./rebar3 as system_test do ct --dir system_test --logdir system_test/logs --config system_test/cfg $(CT_TEST_FLAGS)
 
 system-test-deps: | system_test/aest_hard_fork_SUITE_data/db/mnesia_ae-uat-epoch_backup.tar
-	docker pull aeternity/epoch:v0.11.0
+	docker pull aeternity/epoch:v0.11.1
+	$(info The downloaded DB backup is not checked for freshness.)
```
[the remark this answers is not in the document]
Comment (lucafavatella) [reply Makefile]: It is a make function https://www.gnu.org/software/make/manual/make.html#Functions-That-Control-Make

**Comment by eproxus** [comment]
I haven't looked at the code in detail, but I'm a bit concerned that we're adding hardcoded versions in lots of places. How is this going to scale as we release new versions?

**Comment by lucafavatella** [comment]
> I haven't looked at the code in detail, but I'm a bit concerned that we're adding hardcoded versions in lots of places. How is this going to scale as we release new versions?

@eproxus  I can have a look at removing the versions from the config - though that would make the error in case of misconfiguration less readable. (I would also need to make the aeu_env unit test on the full config example in the test data more strict.)

**Comment by lucafavatella** [comment]
@jur0 I removed verbose logging on node startup as you requested, and force-pushed.

**Comment by lucafavatella** [comment]
> I was also referring to hard coded versions in the tests, like -define(NEW_PROTOCOL_VERSION, 11). for example. What will this file look like once we're ten versions ahead? And how will it be maintained?

@eproxus I minimized usage of hard coded protocol version in test - in new commit I pushed.

File: system_test/aest_hard_fork_SUITE.erl [hunk 2]
```diff
@@ -272,48 +280,36 @@ end_per_group(hard_fork_new_chain_with_tx, Config) ->
...
 end_per_testcase(old_node_can_receive_chain_from_other_old_node, Config) ->
     aest_nodes:ct_cleanup(Config);
-end_per_testcase(restore_db_backup_with_short_chain_on_new_node, _) -> ok;
-end_per_testcase(new_node_accepts_long_old_chain_from_other_new_node_up_to_height_of_new_protocol, _) -> ok;
-end_per_testcase(new_node_can_receive_short_old_chain_from_other_new_node, _) -> ok;
-end_per_testcase(new_node_can_mine_on_old_chain_using_old_protocol, _) -> ok;
-end_per_testcase(new_node_can_mine_on_old_chain_using_new_protocol, _) -> ok;
+end_per_testcase(new_node_accepts_long_old_chain_from_old_node_up_to_height_of_new_protocol, Config) ->
+    aest_nodes:ct_cleanup(Config);
+end_per_testcase(TC, _) when ?is_testcase_in_group_upgrade_flow_smoke_test(TC) -> ok;
```
[the remark this answers is not in the document]
Comment (eproxus) [reply system_test/aest_hard_fork_SUITE.erl]: If you count all test cases in the macro as a separate clause, it is clean that the majority of test cases returns `ok`. I think the implementation should be:
```
end_per_testcase(a, Cfg) -> aest_nodes:ct_cleanup(Cfg);
end_per_testcase(b, Cfg) -> aest_nodes:ct_cleanup(Cfg);
end_per_testcase(_, _) -> ok.
```

File: system_test/aest_hard_fork_SUITE.erl [hunk 3]
```diff
@@ -272,48 +280,36 @@ end_per_group(hard_fork_new_chain_with_tx, Config) ->
...
     aest_nodes:ct_setup(Config);
-init_per_testcase(restore_db_backup_with_short_chain_on_new_node, Config) -> Config;
-init_per_testcase(new_node_accepts_long_old_chain_from_other_new_node_up_to_height_of_new_protocol, Config) -> Config;
-init_per_testcase(new_node_can_receive_short_old_chain_from_other_new_node, Config) -> Config;
-init_per_testcase(new_node_can_mine_on_old_chain_using_old_protocol, Config) -> Config;
-init_per_testcase(new_node_can_mine_on_old_chain_using_new_protocol, Config) -> Config;
+init_per_testcase(new_node_accepts_long_old_chain_from_old_node_up_to_height_of_new_protocol, Config) ->
+    aest_nodes:ct_setup(Config);
+init_per_testcase(TC, Config) when ?is_testcase_in_group_upgrade_flow_smoke_test(TC) -> Config;
 init_per_testcase(new_node_can_mine_spend_tx_on_old_chain_using_old_protocol, Config) -> Config;
```
[the remark this answers is not in the document]
Comment (eproxus) [reply system_test/aest_hard_fork_SUITE.erl]: See [my other comment](#r183694075)

**Comment by lucafavatella** [comment]
@eproxus Addressed.

**Pull request merged**